arcsi/api/show.py

```diff
@@ -88,15 +89,18 @@ def make_show(self, data, **kwargs):
 @arcsi.route("/show", methods=["GET"])
 # We use this route on the legacy for a massive shows query
 @arcsi.route("/show/all", methods=["GET"])
+@auth_required
 def list_shows():
     return shows_schema.dumps(get_shows())
 
 
 @arcsi.route("/show/all_without_items", methods=["GET"])
+@auth_required
 def list_shows_without_items():
     return shows_schedule_schema.dumps(get_shows())   
 
 @arcsi.route("/show/schedule", methods=["GET"])
+@auth_required
```

Comment:
Here however I don't agree with this way of updating all the existing routes. 

This change is more related to the 2nd part of this feature, validation between the Arcsi api and the Arcsi website.

If you would like to test the api tokens and the permissions use 1 dummy copied from the existing apis or better yet just make some new api endpoints (which we might then keep as well) and test those. 

With the current approach, it's hard to figure out the next steps clearly and even harder to think what the ideal solution is going to be, that's why I'd ask for some revision on these.

Reply:
I added these new decorators on the show related APIs in order to demonstrate praetorian's features. I added the auth_required() for the GET requests and roles_required() for POST and DELETE requests, seemed to me as a probably right implementation, but it's still a draft PR, so I'm happy to hear your ideas :)